app/routes.py

```diff
@@ -2,35 +2,110 @@
 from flask_login import login_user, logout_user, current_user, login_required
 from werkzeug.urls import url_parse
 from werkzeug import secure_filename
-from app import app, db
+from app import app, db, crypto, email
 from app.forms import LoginForm, RegistrationForm, VerifyForm, AdminAuthorizationForm
 from app.models import User, PPE, Hospital, Wants, Has, Exchanges, Exchange, EXCHANGE_COMPLETE, EXCHANGE_COMPLETE_TEXT, EXCHANGE_COMPLETE_ADMIN, EXCHANGE_COMPLETE_ADMIN_TEXT, EXCHANGE_COMPLETE_HOSPITAL_CANCELED, EXCHANGE_COMPLETE_HOSPITAL_CANCELED_TEXT, EXCHANGE_COMPLETE_ADMIN_CANCELED, EXCHANGE_COMPLETE_ADMIN_CANCELED_TEXT, EXCHANGE_UNVERIFIED, EXCHANGE_UNVERIFIED_TEXT, EXCHANGE_IN_PROGRESS, EXCHANGE_IN_PROGRESS_TEXT, EXCHANGE_NOT_ACCEPTED, EXCHANGE_ACCEPTED_NOT_SHIPPED, EXCHANGE_ACCEPTED_SHIPPED, EXCHANGE_ACCEPTED_RECEIVED, EXCHANGE_HOSPITAL_CANCELED, EXCHANGE_ADMIN_CANCELED, EXCHANGE_ADMIN_NOT_VERIFIED
 from app import crypto
 from app import email
 from datetime import datetime
-from sqlalchemy import desc
+from sqlalchemy import desc, and_, or_
+from sqlalchemy.orm import contains_eager
 
 import json
 import os
 import math
+import inspect
+
+
+import logging
+logging.basicConfig()
+# logging.getLogger('sqlalchemy.engine').setLevel(logging.INFO)
 
 @app.route('/')
 @app.route('/index')
 @login_required
 def index():
-    user = User.query.filter_by(username=current_user.username).first()
+    user = User.query.\
+        filter_by(username=current_user.username).\
+        join(Hospital).\
+        options(contains_eager(User.hospital)).\
```

Comment:
If you don't do `contains_eager`, it has to do another round-trip to the DB every time you access the joined field. That ends up being really unperformant, so I just did `contains_eager` everywhere for consistency and simplicity.

I don't think this is the _right_ way to solve this problem, but SQLAlchemy's documentation is bad and this works.